specs/lemmas/Spec.Hash.Incremental.fst

```diff
@@ -596,10 +596,291 @@ let md_is_hash_incremental
      }
 #pop-options
 
-let hash_is_hash_incremental (a: hash_alg) (input: bytes { S.length input <= max_input_length a }):
+let sha3_is_incremental1
```

Comment:
should we split this file? e.g., each new file will correspond to one family of hash functions, i.e., blake2, sha2, sha3, etc.

Reply:
sure, I can do that, it might require me to friend a module or two, but ok!